### PR TITLE
Add service connection from Testcontainers OpenTelemetry Collector

### PR DESCRIPTION
**Pull request by eddumelendez**

### CODE REVIEW

File: spring-boot-project/spring-boot-testcontainers/src/test/java/org/springframework/boot/testcontainers/service/connection/otlp/OtlpContainerConnectionDetailsFactoryIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.matchesPattern;
+
+/**
+ * Tests for {@link OpenTelemetryConnectionDetailsFactory}.
+ *
+ * @author Eddú Meléndez
+ * @author Jonatan Ivanov
```
Comment (eddumelendez) [inline spring-boot-project/spring-boot-testcontainers/src/test/java/org/springframework/boot/testcontainers/service/connection/otlp/OtlpContainerConnectionDetailsFactoryIntegrationTests.java:53]: Adding Jonatan due to the test was the one he submitted in micrometer https://github.com/micrometer-metrics/micrometer/blob/main/implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/OTelCollectorIntegrationTest.java

**Review by jonatan-ivanov — COMMENTED** [review]
We've recently added support for `OtlpSpanExporter` too. Do you think it would worth using `OtlpConnectionDetails` there too?

See: https://github.com/spring-projects/spring-boot/pull/34508

**Comment by eddumelendez** [comment]
@jonatan-ivanov I think it make sense 😃 PR updated

**Comment by mhalbritter** [comment]
Hey, I was just looking at that PR, sorry for taking my time. Do you mind updating it to the latest `main` and fixing the conflicts? Maybe we can get this into 3.2.0-M3.

**Comment by eddumelendez** [comment]
Hi @mhalbritter, I'll 👍🏽

**Comment by eddumelendez** [comment]
PR updated

**Comment by mhalbritter** [comment]
Thanks!

**Comment by mhalbritter** [comment]
Thank you very much @eddumelendez !

**Pull request closed without merging**